src/engine/surface_load.c

```diff
@@ -205,11 +205,11 @@ static s32 upper_cell_index(s32 coord) {
     //! Some wall checks are larger than the buffer, meaning wall checks can
     //  miss walls that are near a cell border.
     if (coord % CELL_SIZE > CELL_SIZE - 50) {
-        index += 1;
+        index++;
     }
 
-    if (index > NUM_CELLS_INDEX) {
-        index = NUM_CELLS_INDEX;
+    if (index > (NUM_CELLS - 1)) {
+        index = (NUM_CELLS - 1);
```

Comment:
why remove `NUM_CELLS_INDEX`? I prefer that define because the `-1` can seem random